### ISSUE TITLE
sometimes, renku branch is not found. unclear why. currently happens for default isgri request

### ISSUE THREAD

**Issue by volodymyrss** [oda-hub/dispatcher-app#385]

**Comment by burnout87** [comment]
Is the branch actually created? Could you verify that?

**Comment by volodymyrss** [comment]
the branch is not found because it's not created. That's what I meant. Sorry if unclear

**Comment by burnout87** [comment]
Since you say that it happens with isgri default request, it sounds like a request that has already been attempted in the past, and for which the branch was created...but again, it should force-push also on existing branches

**Comment by volodymyrss** [comment]
there is another essential problem, which maybe contributes here:
jobid is the same for all requests in one session. It means that it is often wrong. 
Could you check if you reproduce this?
* make default ISGRI request. open on renku
* switch tab, make default spi-acs request. open on renku
* observe that branch is the same in both cases

**Comment by burnout87** [comment]
Ok I will try it, but the more I think about it the more I think I see where the error might come from.

**Comment by burnout87** [comment]
Just tried, the two requests generate the same `job_id`, therefore, they currently end-up on the same renku-branch.

Perhaps we could use also the instrument and product_type for the branch name.

**Comment by volodymyrss** [comment]
> Just tried, the two requests generate the same `job_id`, therefore, they currently end-up on the same renku-branch.
> 
> Perhaps we could use also the instrument and product_type for the branch name.

No, it should impossible that two different requests generate the same job id. instrument and product_type are included in job id.
Probably wrong job id is being sent to push-branch.

**Comment by burnout87** [comment]
Ok yes that's true, then the problem is in the frontend, I will take a look since I developed that part that makes the request

**Comment by burnout87** [comment]
Confirm the issue on the frontend. I will open a dedicated PR.

I plan also to fix a graphical (mis)behavior
